lib/resty/websocket/client.lua

```diff
@@ -98,9 +98,17 @@ function _M.connect(self, uri, opts)
         path = "/"
     end
 
-    local ssl_verify, proto_header, origin_header, sock_opts = false
+    local ssl_verify, proto_header, origin_header, custom_header, sock_opts = false
 
     if opts then
+        local headers = opts.headers
+        if headers then
+            custom_header = ""
+            for field,value in pairs(headers) do
```

Comment:
@edubart The lua-resty-http library you referenced is not our work, so we do not really care about being consistent with all the 3rd-party libraries. The upcoming `ngx.req.get_header_list()` and `ngx.resp.get_header_list()` API functions of the OpenResty core will use a similarly structured flat array. OpenResty's core APIs and libraries care more about performance.